src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/TestCaseCachingService.kt

```diff
@@ -161,6 +161,26 @@ class TestCaseCachingService {
             )
         }
 
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
+
+            other as CachedCompactTestCase
+
+            if (testName != other.testName) return false
+            if (testCode != other.testCode) return false
+            if (coveredLines != other.coveredLines) return false
+
+            return true
+        }
+
+        override fun hashCode(): Int {
+            var result = testName.hashCode()
+            result = 31 * result + testCode.hashCode()
```

Comment:
Is there any specific reason for `31`?

Reply:
This is auto-generated code, having 31 here makes the hash function better (not sure what the exact math here is, this is what intellij generates).